src/main/java/org/eclipse/cargotracker/interfaces/booking/sse/RealtimeCargoTrackingService.java

```diff
@@ -53,8 +51,10 @@ public void close() {
   }
 
   public void onCargoUpdated(@ObservesAsync @CargoUpdated Cargo cargo) {
-    logger.log(Level.FINEST, "SSE event broadcast for cargo: {0}", cargo);
-    broadcaster.broadcast(cargoToSseEvent(cargo));
+    if (sse.newEventBuilder() != null) {
```

Comment:
Shouldn't this be just checking if the broadcaster is set? Why is the SSE check needed? This looks odd. If there is a specific reason, it should be explained via comment.

Reply:
I'll add some comments regarding why these checks are needed in my next commit.

Reply:
I really need to check the specification on some of this. The project is supposed to provide good examples of Jakarta EE code. If what we are doing here is working around a Liberty bug and introducing strange code because of that, we really can't  accept this PR and have to defer Liberty support.